### PR TITLE
FHIR-41242 Criteria Reference Extension needs to be more generic and reusable across IGs

### PR DESCRIPTION
**Pull request by bryantaustin13**
Goes along with PR https://github.com/HL7/fhir-extensions/pull/57#issue-2051113255 which contains the actual extension. This PR uses a substitute StructureDefinition: StructureDefinition-external-extension-criteriaReference.json

### CODE REVIEW

File: input/examples/Bundle-single-gaps-closed-indv-report01.json [hunk 1]
```diff
@@ -225,11 +225,11 @@
           {
             "extension": [
               {
-                "url": "http://hl7.org/fhir/us/davinci-deqm/StructureDefinition/extension-criteriaReference",
+                "url": "http://hl7.org/fhir/StructureDefinition/criteriaReference",
```
Comment (p9g) [inline input/examples/Bundle-single-gaps-closed-indv-report01.json:228]:
```suggestion
                "url": "http://hl7.org/fhir/StructureDefinition/cqf-criteriaReference",
```

File: input/examples/Bundle-single-gaps-closed-indv-report01.json [hunk 2]
```diff
@@ -225,11 +225,11 @@
...
                 "valueString": "initial-population"
-              }, 
+              },
               {
-                "url": "http://hl7.org/fhir/us/davinci-deqm/StructureDefinition/extension-criteriaReference",
+                "url": "http://hl7.org/fhir/StructureDefinition/criteriaReference",
```
Comment (p9g) [inline input/examples/Bundle-single-gaps-closed-indv-report01.json:232]:
```suggestion
                "url": "http://hl7.org/fhir/StructureDefinition/cqf-criteriaReference",
```

File: input/examples/Bundle-single-gaps-closed-indv-report01.json [hunk 3]
```diff
@@ -238,11 +238,11 @@
           {
             "extension": [
               {
-                "url": "http://hl7.org/fhir/us/davinci-deqm/StructureDefinition/extension-criteriaReference",
+                "url": "http://hl7.org/fhir/StructureDefinition/criteriaReference",
```
Comment (p9g) [inline input/examples/Bundle-single-gaps-closed-indv-report01.json:241]:
```suggestion
                "url": "http://hl7.org/fhir/StructureDefinition/cqf-criteriaReference",
```

File: input/examples/Bundle-single-gaps-closed-indv-report01.json [hunk 4]
```diff
@@ -238,11 +238,11 @@
...
                 "valueString": "initial-population"
               },
               {
-                "url": "http://hl7.org/fhir/us/davinci-deqm/StructureDefinition/extension-criteriaReference",
+                "url": "http://hl7.org/fhir/StructureDefinition/criteriaReference",
```
Comment (p9g) [inline input/examples/Bundle-single-gaps-closed-indv-report01.json:245]:
```suggestion
                "url": "http://hl7.org/fhir/StructureDefinition/cqf-criteriaReference",
```

**Comment by p9g** [comment]
I started making comments in line, but rather than do each one, in general, change
`"url": "http://hl7.org/fhir/StructureDefinition/criteriaReference"`
to
`"url": "http://hl7.org/fhir/StructureDefinition/cqf-criteriaReference"`
see https://hl7.org/fhir/extensions/5.1.0/StructureDefinition-cqf-criteriaReference.html

**Review by brynrhodes — APPROVED** [review]

**Pull request merged**